.azure-pipelines/windows.sh

```diff
@@ -102,31 +111,34 @@ cd "$DMD_DIR/../phobos"
 
 cd "$DMD_DIR/compiler/test"
 
-# build run.d testrunner and its tools while host compiler is untampered
-if [ "$MODEL" == "32omf" ] ; then
-    TOOL_MODEL=32;
-else
-    TOOL_MODEL="$MODEL"
-fi
-
-"$HOST_DC" -m$TOOL_MODEL -g -i run.d
-./run tools
-
 # Rebuild dmd with ENABLE_COVERAGE for coverage tests
 if [ "${DMD_TEST_COVERAGE:-0}" = "1" ] ; then
 
     # Recompile debug dmd + unittests
     rm -rf "$DMD_DIR/generated/windows"
-    DFLAGS="-L-LARGEADDRESSAWARE" ../../generated/build.exe --jobs=$N ENABLE_DEBUG=1 ENABLE_COVERAGE=1 dmd
-    DFLAGS="-L-LARGEADDRESSAWARE" ../../generated/build.exe --jobs=$N ENABLE_DEBUG=1 ENABLE_COVERAGE=1 unittest
+    ../../generated/build.exe -j$N MODEL=$TOOL_MODEL DFLAGS="-L-LARGEADDRESSAWARE" ENABLE_DEBUG=1 ENABLE_COVERAGE=1 dmd
+    ../../generated/build.exe -j$N MODEL=$TOOL_MODEL DFLAGS="-L-LARGEADDRESSAWARE" ENABLE_DEBUG=1 ENABLE_COVERAGE=1 unittest
 fi
 
+"$HOST_DC" -m$TOOL_MODEL -g -i run.d
+
 if [ "$MODEL" == "32omf" ] ; then
+    # Pre-build the tools while the host compiler's sc.ini is untampered (see below).
+    ./run tools
+
     # WORKAROUND: Make Optlink use freshly built Phobos, not the host compiler's.
-    # Optlink apparently prefers LIB in sc.ini over the LIB env variable (and
-    # `-conf=` for DMD apparently doesn't prevent that, and there's apparently
-    # no sane way to specify a libdir for Optlink in the DMD cmdline).
+    # Optlink apparently prefers LIB in sc.ini (in the same dir as optlink.exe)
+    # over the LIB env variable (and `-conf=` for DMD apparently doesn't prevent
+    # that, and there's apparently no sane way to specify a libdir for Optlink
+    # in the DMD cmdline either).
     rm "$DMD_DIR/tools/dmd2/windows/bin/sc.ini"
+    # We also need to remove LIB from the freshly built compiler's sc.ini -
+    # not all test invocations use `-conf=`.
+    sed -i 's|^LIB=.*$||g' "$DMD_DIR/generated/windows/release/$TOOL_MODEL/sc.ini"
+    # Okay, now the lib directories are controlled by the LIB env variable.
+    # run.d prepends the dir containing freshly built phobos.lib; we still need
+    # the DMC and Windows libs from the host compiler.
+    export LIB="$DMD_DIR/tools/dmd2/windows/lib"
```

Comment:
OMG, that was a rabbit hole again. In short, I'm not sure how exactly it has been working so far without adapting the generated `sc.ini`, but this was required to get the 32-bit OMF CI job working again when using `build.d` directly.